### DISCUSSION TITLE
replications uneven distribution

### DISCUSSION THREAD

**Discussion by vivawz** [milvus-io/milvus#37349] [Q&A and General discussion]
There are 20 querynodes,  my collection has 2 segments. Here's the infos:

[segmentID: 453399159701083856
collectionID: 453399159756474075
partitionID: 453399159756474076
num_rows: 401400
index_name: "embeddings"
indexID: 453399159757276564
state: Sealed
nodeIds: 4982
nodeIds: 5023
nodeIds: 4992
nodeIds: 5027
nodeIds: 4989
, segmentID: 453399159701083855
collectionID: 453399159756474075
partitionID: 453399159756474076
num_rows: 408047
index_name: "embeddings"
indexID: 453399159757276564
state: Sealed
nodeIds: 4982
nodeIds: 4984
nodeIds: 4992
nodeIds: 4989
nodeIds: 5026
]

for example, 4982 has both the 2 segments.  so It has high load and use more cpu

**Comment by yhmo** [comment]
Did you set the `num_replica ` to 5 when you call collection.load()?
There are more collections in this cluster, right?

**Reply by vivawz** [reply]
yes,  I set num_replica=5.  And there are more collections.  But I think the segments of the same collection should put to different nodes. otherwise cpu usage will very diff and cause some problems

**Reply by yhmo** [reply]
If the cluster has only one collection, I believe the two segments are loaded to different nodes. But since there are more segments of other collections. the balancer is not only for one collection.
